borg/repository.py

```diff
@@ -188,10 +191,23 @@ def commit(self, save_space=False):
         self.write_index()
         self.rollback()
 
-    def open_index(self, transaction_id):
+    def open_index(self, transaction_id, auto_recover=True):
         if transaction_id is None:
             return NSIndex()
-        return NSIndex.read((os.path.join(self.path, 'index.%d') % transaction_id).encode('utf-8'))
+        index_path = (os.path.join(self.path, 'index.%d') % transaction_id).encode('utf-8')
```

Comment:
if you move "% transaction_id" directly right to 'index.%d', you don't need the outer parens.